integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py

```diff
@@ -34,41 +37,48 @@ def __init__(
         self,
         *,
         api_key: Secret = Secret.from_env_var("PINECONE_API_KEY"),  # noqa: B008
-        environment: str = "us-west1-gcp",
         index: str = "default",
         namespace: str = "default",
         batch_size: int = 100,
         dimension: int = 768,
-        **index_creation_kwargs,
+        spec: Optional[Dict[str, Any]] = None,
+        metric: Literal["cosine", "euclidean", "dotproduct"] = "cosine",
```

Comment:
- `environment` no longer exists in Pinecone API
- `spec` has been introduced in Pinecone API
- in our implementation, `index_creation_kwargs` allowed passing `metric` and other args now incorporated in `spec`